Automattic/woocommerce-payments
# feat: add Store API multi-currency support

Fixes #

#### Changes proposed in this Pull Request









Adding multi-currency support to the Store API.
Tangentially, this also helps fix the multi-currency feature with the tokenized cart.

#### Testing instructions





The main purpose of this is to support the multi-currency feature with the Store API.
You can check this behavior by going into a block-based page (e.g.: block-based checkout page) and making this example request on a store with USD as the main currency and with EUR enabled:
```js
window.wp.apiFetch( {
    method: 'GET',
    path: '/wc/store/v1/cart?currency=EUR',
    // omitting credentials, to create a new cart object separate from the user's cart.
    credentials: 'omit',
} )
```

You'll notice that in the response, the `totals.currency_code` string corresponds to the EUR currency.

To further test this behavior with the tokenized cart with PRBs:
- Enable the "Enable Cart-Token implementation for PRBs" flag in your dev tools (you'll need to update them)
- Ensure you have Google Pay/Apple Pay enabled in the merchant's settings
- Ensure you have the multi-currency feature enabled, with a few currencies enabled (e.g.: USD & EUR)
- Ensure the selected currency is not the store's "default" currency (i.e.: as a shopper, select EUR if your store's default currency is USD)
- Go to a product page on your site
- Increase the product's quantity (so that the cached data from the backend is ignored)
- Click on the PRB button, so that the dialog opens
- The price of the product should match the price displayed on the PRB dialog


With this product (the single unit price is 14EUR/15USD):
<img width="886" alt="Screenshot 2024-05-14 at 9 35 18 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/750a1482-679d-43a6-a218-89e08fcc8a31">

Before - selected currency is EUR, but displayed currency in the PRB dialog is USD & price is 2x USD single unit price:
![Untitled](https://github.com/Automattic/woocommerce-payments/assets/273592/5afe9bb1-50d2-48fc-bf91-e288287196d9)


After - selected currency is EUR, displayed currency in the PRB dialog is accurate and price is accurate:
<img width="1169" alt="Screenshot 2024-05-14 at 9 38 31 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/a6799e73-0ae3-4810-b97d-29cb27cbb76b">


-------------------


**Post merge**
